src/providers/ad/ad_gpo_child.c

```diff
@@ -554,10 +556,38 @@ copy_smb_file_to_gpo_cache(SMBCCTX *smbc_ctx,
     errno = 0;
     file = smbc_getFunctionOpen(smbc_ctx)(smbc_ctx, smb_uri, O_RDONLY, 0755);
     if (file == NULL) {
```

Comment:
I wonder if `errno == ENOENT` should be checked as well to avoid calling `smbc_getFunctionOpen()` twice if there is a different error? Unfortunately this is not documented for `smbc_getFunctionOpen()` itself but only for `smbc_open()` which is a wrapper with a global context. But I assume it can still safely being used here.

Reply:
I have to rely on your experience here. I vaguely remember to spent much time on this hardly documented function and examining other applications how they use it. 
The focus was really on making ad child tolerant for case intensive sysvol folder structures, which caused us headaches.  Only checking for Null file handle worked fine since then.
What does libsamba indicate with errno = ENOENT?

Reply:
Hi,

please see the description of `smbc_open()` e.g. at https://gitlab.com/samba-team/samba/-/blob/master/source3/include/libsmbclient.h#L1362 . It is set at https://gitlab.com/samba-team/samba/-/blob/master/source3/libsmb/libsmb_file.c#L121 if `cli_resolve_path()` fails.

bye,
Sumit